src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -360,43 +393,98 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
+        vacationRequestStatus: {
+          ...selectedStatusObject,
+          status: newStatus,
+          vacationRequestId: selectedStatusObject.vacationRequestId,
+          createdAt: selectedStatusObject.createdAt,
+          createdBy: selectedStatusObject.createdBy,
+          message: selectedStatusObject.message,
+          updatedAt: new Date(),
+          updatedBy: person?.keycloakId
+        }
+      });
+
+      const update = latestRequestStatuses.map(requestStatus => (requestStatus.id !== selectedStatusObject.id ? requestStatus : updatedRequestStatus));
+      setLatestRequestStatuses(update);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
     }
   };
 
-  // /**
-  //  * Handle request status
-  //  *
-  //  * @param requestStatus Vacation request status
-  //  */
-  // const handleRequestStatus = (requestStatus: VacationRequestStatuses) => {
-  //   const statusMap = {
-  //     [VacationRequestStatuses.PENDING]: strings.vacationRequests.pending,
-  //     [VacationRequestStatuses.APPROVED]: strings.vacationRequests.approved,
-  //     [VacationRequestStatuses.DECLINED]: strings.vacationRequests.declined
-  //   };
-
-  //   return statusMap[requestStatus] || "";
-  // };
+  /**
+   * Handle request status on click
+   * @param approved is status approved or not
+   * @param requestId the id of vacation request
+   */
+  const handleClick = (approved: boolean, requestId: string | undefined) => {
+    const selectedStatusIndex = latestRequestStatuses.findIndex(s => s.vacationRequestId === requestId);
+    const selectedStatusObject = latestRequestStatuses.find(s => s.vacationRequestId === requestId);
+    if (approved && selectedStatusObject?.status && myRef.current) {
+      selectedStatusObject.status = VacationRequestStatuses.APPROVED;
+      myRef.current.className = "approved";
+      updateVacationRequestStatus(selectedStatusObject, selectedStatusObject.status);
+    }
+    if (!approved && selectedStatusObject?.status && myRef.current) {
+      selectedStatusObject.status = VacationRequestStatuses.DECLINED;
+      myRef.current.className = "declined";
+      updateVacationRequestStatus(selectedStatusObject, selectedStatusObject.status);
+    }
+    latestRequestStatuses[selectedStatusIndex] = selectedStatusObject!;
+
+    if (approvalChanged) {
+      setApprovalChanged(false);
+    } else {
+      setApprovalChanged(true);
+    }
+  };
+
+  /**
+   * Interface for StatusTableCell props
+   */
+  interface StatusTableCellProps {
+    requestStatusId: string | undefined,
+    requestStatus: VacationRequestStatuses,
+    approvalChanged: boolean
+  }
+  
+  /**
+   * Handle request status on click
+   *
+   * @param type status component
+   */
+  function StatusTableCell(props: StatusTableCellProps) {
+    const { requestStatusId, requestStatus } = props;
+    const [ statusText, setStatusText ] = useState<string>(getLocalizedRequestStatus(requestStatus));
```

Comment:
I suspect the <string> type is not needed here as typescript can infer the type from the passed param, but check

Reply:
Type was, indeed, not needed.


```suggestion
    const [ statusText, setStatusText ] = useState(getLocalizedRequestStatus(requestStatus));
```